rewrite-gradle/src/test/java/org/openrewrite/gradle/util/GradleWrapperTest.java

```diff
@@ -17,14 +17,18 @@
 
 import org.junit.jupiter.api.Test;
 import org.openrewrite.InMemoryExecutionContext;
+import org.openrewrite.Validated;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class GradleWrapperTest {
 
     @Test
     void wrapper() {
-        GradleWrapper gw = GradleWrapper.validate(new InMemoryExecutionContext(), "7.x", "bin", null).getValue();
+        Validated validated = GradleWrapper.validate(new InMemoryExecutionContext(), "7.x", "bin", null);
+        assertThat(validated.isValid()).isTrue();
```

Comment:
This is not sufficient to prove that the recipe works correctly when supplied with the string "bin" - which does not exactly match the enum "Bin"

Reply:
added two most tests to cover enum `All` and null.
if the enum matching succeeds, it will be reflected in the suffix of the download URL. so added download URL verifications in the test.

Reply:
Ah, that's fine. What I intended by that comment is to test the recipe itself with "bin", [like this](https://github.com/openrewrite/rewrite/pull/2751/commits/20cb2841991f4aa8c307963e58e824474c8832a8)